src/runtime/exceptions.cs

```diff
@@ -181,9 +200,19 @@ internal static void SetArgsAndCause(IntPtr ob)
 
             if (e.InnerException != null)
             {
-                // Note: For an AggregateException, InnerException is only the first of the InnerExceptions.
-                IntPtr cause = CLRObject.GetInstHandle(e.InnerException);
-                Marshal.WriteIntPtr(ob, ExceptionOffset.cause, cause);
+                if(e.InnerException is PythonException)
+                {
+                    // If the error is a Python exception, write the real one
+                    var pyerr =  (e.InnerException as PythonException);
+                    Runtime.XIncref(pyerr.PyValue);
```

Comment:
You might need to call `PythonException.Normalize`. Some python exceptions don't have `Value`

Reply:
I'll revert that part of the code, the call to Normalize makes the python (but not the C# side) exception leak. Probably a variation of the same problem @filmor found with the pyValue of PythonException.